http://www.nntp.perl.org/group/perl.perl5.porters/2006/01/msg108750.html

The following comes from http://rt.cpan.org/Public/Bug/Display.html?id=22506

Still segfaults in assign() on t/tiedhash.t -- SvRV(root) starts returning
null after a call to sv_isobject(root); this doesn't seem right and the code
later doesn't expect it.


On Mon Oct 23 11:25:30 2006, SMPETERS wrote:
> A missing pair of parens causes Template-Toolkit 2.15 to fail to compile
> with bleadperl.  The following patch fixes the problem.
> 

Actually, there is also some test code that fails an assertion, so, the
following patch fixes the compilation and the test failures.

--- xs/Stash.xs.old     2007-01-09 20:27:49.000000000 -0600
+++ xs/Stash.xs 2007-01-09 22:06:30.000000000 -0600
@@ -499,7 +499,7 @@
         }
         
         /* drop-through if not an object or method not found  */
-        switch SvTYPE(SvRV(root)) {
+        switch (SvTYPE(SvRV(root))) {
             
         case SVt_PVHV:				    /* HASH */
             roothv = (HV *) SvRV(root);
@@ -991,7 +991,7 @@
     STRLEN jlen;
     char *joint;

-    if ((svp = av_fetch(args, 0, FALSE)) != NULL) {
+    if (args && (svp = av_fetch(args, 0, FALSE)) != NULL) {
         joint = SvPV(*svp, jlen);
     } else {
         joint = " ";
